### ISSUE TITLE
Heterogeneous Unsupervised Graph Representation Learning

### ISSUE THREAD

**Issue by rusty1s** [pyg-team/pytorch_geometric#3163]
An example of applying Deep Graph Infomax on heterogeneous graphs.

**Comment by Yonggie** [comment]
I'm doing a paper that is exactly `Heterogeneous Unsupervised Graph Representation Learning` method. Hope it to be accepted as one example method in the project. But it's not finished yet, still working in progress.

**Comment by rusty1s** [comment]
Of course! Let me know once it is finished. We are very interested in adding such an example to the repository.

**Comment by Yonggie** [comment]
I just created a PR to add hetero unsupervised method simple example with DGI method applied. You may check it out.

**Comment by Yonggie** [comment]
yeah, I just forked the lasted version and submitted a new PR. How about this time? All checked.

**Comment by rusty1s** [comment]
Looks good. Thank you!

**Comment by Yonggie** [comment]
Sure, with pleasure.

**Comment by Yonggie** [comment]
Hi, new PR's ready for review.
